applicationset/services/pull_request/bitbucket_cloud.go

```diff
@@ -73,6 +73,23 @@ func NewBitbucketCloudServiceBasicAuth(username, password, baseUrl, owner, repos
 	}, nil
 }
 
+func NewBitbucketCloudServiceBearerToken(baseUrl, owner, repositorySlug string, bearerToken string) (PullRequestService, error) {
+	url, err := parseUrl(baseUrl)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing base url of %s for %s/%s: %v", baseUrl, owner, repositorySlug, err)
+	}
+
+	// There is currently no method to explicitly not require auth
```

Comment:
```suggestion
```

This comment was only in `NewBitbucketCloudServiceNoAuth` to explain why we're calling `NewOAuthbearerToken` with an empty string. Not relevant in this case so think it can be removed.

Reply:
Thanks for the review. While applying your comment I've noticed that there's redundancy in my pull request, `NewBitbucketCloudServiceNoAuth` and `NewBitbucketCloudServiceBearerToken` were almost the same. So I've cleaned this up: https://github.com/Jamess-Lucass/argo-cd/pull/2/files#diff-8523234aa497230c1a958d497fd318024e3659df83f15163537f2a4f98a43e77L76 
Now `NewBitbucketCloudServiceNoAuth` uses `NewBitbucketCloudServiceBearerToken` to operate.